docker/docker.go

```diff
@@ -1,147 +1,73 @@
 package docker
 
 import (
+	"github.com/alaa/pencil-go/registry"
 	docker "github.com/fsouza/go-dockerclient"
-	"strings"
-	"time"
 )
 
-const (
-	Endpoint = "unix:///var/run/docker.sock"
-	Interval = 5 * time.Second
-)
-
-const SRV_NAME = "SRV_NAME"
-
-type DockerClient interface {
+type dockerClient interface {
 	ListContainers(opts docker.ListContainersOptions) ([]docker.APIContainers, error)
 	InspectContainer(id string) (*docker.Container, error)
 }
 
-type TCPPorts []string
-
-type UDPPorts []string
-
-type ContainerInfo struct {
-	ID          string
-	Name        string
-	ImageID     string
-	ImageName   string
-	ServiceName string
-	Env         map[string]string
-	Created     time.Time
-	Config      *docker.Config
-	TCPPorts    TCPPorts
-	UDPPorts    UDPPorts
-}
-
-type EnvVariables map[string]string
-
-type DockerAdapter struct {
-	dockerClient DockerClient
+// ContainerRepository is docker-based implementation of registry.ContainerRepository
+type ContainerRepository struct {
+	dockerClient dockerClient
 }
 
-func NewDockerAdapter(dockerClient DockerClient) *DockerAdapter {
-	return &DockerAdapter{dockerClient: dockerClient}
+// NewContainerRepository creates new instance of ContainerRepository structure
+func NewContainerRepository(dockerClient dockerClient) *ContainerRepository {
+	return &ContainerRepository{dockerClient: dockerClient}
 }
 
-// GetRunningContainers finds running containers and returns specific details.
-// TOOD package should return error, logging should be disabled.
-func (d *DockerAdapter) GetRunningContainers() ([]ContainerInfo, error) {
-	containersIDs, err := d.getContainersIDs()
+// GetAll returns list of all running docker containers
+func (cr *ContainerRepository) GetAll() []registry.Container {
+	containersIDs, err := cr.getContainersIDs()
 	if err != nil {
-		return []ContainerInfo{}, err
```

Comment:
Here you silently swallow the error. When something is wrong it will be hard to detect what exactly is wrong. Can you at least add logging for these "swallowed" errors?